build.gradle

```diff
@@ -377,6 +398,7 @@ project ('bindings') {
 project('segmentstore:server') {
     dependencies {
         compile project(':common')
+        compile project(':common_server')
```

Comment:
I am not sure that using the underscore is a good practice.
What about common.server?

Reply:
That was my first choice too. However Gradle completely refused to accept that. Having `common` and `common:server` (which means `common:server` must be in a direct sudirectory of `common`) prevented me from being able to access the packages in `common:server` (that is, when adding a reference in `segmentstore:server` to `common:server`, Gradle complained about a circular dependency and Intellij Idea refused to see those packages. Moving this code to a top-level package (like it is now) did not have this issues.

I couldn't figure out what was wrong there since I did nothing special. We do use that pattern before (`segmentstore:server` and `segmentstore:server:host`) albeit with slight differences - the only difference I could tell was the both `common` and `common:server` share the same root package name `io.pravega.common`. I didn't want to create new/sub packages for this code.